src/realm/sync/changeset.cpp

```diff
@@ -10,48 +10,32 @@ using namespace realm;
 using namespace realm::sync;
 using namespace realm::util;
 
-Changeset::Changeset()
-{
-    m_strings = std::make_shared<InternStrings>();
-    m_string_buffer = std::make_shared<std::string>();
-}
-
-Changeset::Changeset(const Changeset& other, share_buffers_tag)
-    : version(other.version)
-    , last_integrated_remote_version(other.last_integrated_remote_version)
-    , origin_timestamp(other.origin_timestamp)
-    , origin_file_ident(other.origin_file_ident)
-{
-    m_strings = other.m_strings;
-    m_string_buffer = other.m_string_buffer;
-}
-
 InternString Changeset::intern_string(StringData str)
 {
     if (InternString interned = find_string(str))
         return interned;
 
-    REALM_ASSERT(m_string_buffer->size() < std::numeric_limits<uint32_t>::max());
-    REALM_ASSERT(m_strings->size() < std::numeric_limits<uint32_t>::max());
+    REALM_ASSERT(m_string_buffer.size() < std::numeric_limits<uint32_t>::max());
+    REALM_ASSERT(m_strings.size() < std::numeric_limits<uint32_t>::max());
     REALM_ASSERT(str.size() < std::numeric_limits<uint32_t>::max());
 
     // FIXME: Very slow.
```

Comment:
I didn't notice either of these FIXMEs being actual perf problems in the things I looked at, but I don't know if that's because the FIXMEs are wrong/out-of-date or if I just didn't test the scenario where they would be problems.